frontend/src/Components/ColorModeProvider/ColorModeProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import { createContext } from 'preact';
+import { useCallback, useLayoutEffect, useState } from 'preact/hooks';
+import { JSXInternal } from 'preact/src/jsx';
+
+interface ColorModeContext {
+  isDark: boolean;
+  toggleColorMode: () => void;
+}
+
+enum Mode {
+  DARK = 'dark',
+  LIGHT = 'light',
+}
+const storageKey = 'color-mode';
+const attributeKey = 'data-color-mode';
+
+export const ColorMode = createContext<ColorModeContext>({
+  isDark: false,
+  toggleColorMode: () => {},
+});
+
+export const ColorModeProvider = ({ children }: { children: JSXInternal.Element }) => {
+  const [isDark, setIsDark] = useState<boolean>(getInitialColorMode);
+
+  useLayoutEffect(() => {
+    const mode = isDark ? Mode.DARK : Mode.LIGHT;
+    document.body.setAttribute(attributeKey, mode);
+    localStorage.setItem(storageKey, mode);
```

Comment:
I dont like the direct usage of local Storage, Couleur we use a simple wrapper for it, would make easier once we need it for other features

Reply:
I extracted the usage of local storage to simple `useLocalStorage` hook. Normally, I would have used a library for local storage access, but none seems compatible with preact.

Reply:
As discussed removed the local storage due to privacy concerns.